### PR TITLE
More categorical tests and disable shap sparse test.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

File: tests/python-gpu/test_gpu_prediction.py [hunk 1]
```diff
@@ -212,6 +212,9 @@ def test_shap(self, num_rounds, dataset, param):
            tm.dataset_strategy, shap_parameter_strategy)
     @settings(deadline=None, max_examples=20)
     def test_shap_interactions(self, num_rounds, dataset, param):
+        if dataset.name == 'sparse':
+            return
```
Comment (hcho3) [inline tests/python-gpu/test_gpu_prediction.py:216]: Can we use `pytest.xfail` or `pytest.skip` here? This way, pytest will show that this test was skipped and the reason why.

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: tests/python-gpu/test_gpu_prediction.py (hunk 1 above)
Comment (trivialfis) [reply tests/python-gpu/test_gpu_prediction.py]: Done.

**Comment by hcho3** [comment]
Please specify the `reason` argument in `pytest.xfail()`. This way, pytest will report why the test was skipped.

**Comment by trivialfis** [comment]
Done:
```
========================================================================================= short test summary info =========================================================================================
XFAIL tests/python-gpu/test_gpu_prediction.py::TestGPUPredict::test_shap_interactions
  reason: GPU shap with sparse is flaky: https://github.com/dmlc/xgboost/issues/6074
```

**Review by hcho3 — APPROVED** [review]

**Comment by hcho3** [comment]
Grr... MSVC is acting up again:

> LINK : fatal error LNK1000: Internal error during ParseCommandLine [C:\Users\Administrator\workspace\xgboost-win64_PR-6219\build\src\objxgboost.vcxproj]

Let me restart the Windows machine.

**Pull request merged**